plugins/modules/idrac_license.py

```diff
@@ -0,0 +1,824 @@
+from ansible.module_utils.six.moves.urllib.error import URLError, HTTPError
+from ansible.module_utils.urls import ConnectionError, SSLValidationError
+from ansible_collections.dellemc.openmanage.plugins.module_utils.utils import (
+    get_idrac_firmware_version, get_dynamic_uri, get_manager_res_id,
+    validate_and_get_first_resource_id_uri, remove_key, idrac_redfish_job_tracking)
+
+REDFISH = "/redfish/v1"
+MANAGERS_URI = "/redfish/v1/Managers"
+IDRAC_JOB_URI = "{res_uri}/Jobs/{job_id}"
+
+OEM = "Oem"
+MANUFACTURER = "Dell"
+LICENSE_MANAGEMENT_SERVICE = "DellLicenseManagementService"
+ACTIONS = "Actions"
+EXPORT_LOCAL = "#DellLicenseManagementService.ExportLicense"
+EXPORT_NETWORK_SHARE = "#DellLicenseManagementService.ExportLicenseToNetworkShare"
+IMPORT_LOCAL = "#DellLicenseManagementService.ImportLicense"
+IMPORT_NETWORK_SHARE = "#DellLicenseManagementService.ImportLicenseFromNetworkShare"
+
+INVALID_LICENSE_MSG = "License id '{license_id}' is invalid."
+SUCCESS_EXPORT_MSG = "Successfully exported the license."
+SUCCESS_DELETE_MSG = "Successfully deleted the license."
+SUCCESS_IMPORT_MSG = "Successfully imported the license."
+FAILURE_MSG = "Unable to '{operation}' the license with id '{license_id}'."
+FAILURE_IMPORT_MSG = "Unable to import the license."
+NO_FILE_MSG = "License file not found."
+INVALID_FILE_MSG = "File extension is invalid. Supported extensions for local 'share_type' " \
+                   "is: .txt and .xml, and for network 'share_type' is: .xml."
+INVALID_DIRECTORY_MSG = "Provided directory path '{path}' is not valid."
+INSUFFICIENT_DIRECTORY_PERMISSION_MSG = "Provided directory path '{path}' is not writable. " \
+                                        "Please check if the directory has appropriate permissions"
+MISSING_PARAMETER_MSG = "Missing required parameter 'file_name'."
+
+IGNORE_CERTIFICATE_WARNING = {"off": "Off", "on": "On"}
+PROXY_SUPPORT = {"off": "Off", "default_proxy": "DefaultProxy", "parameters_proxy": "ParametersProxy"}
+PROXY_TYPE = {"http": "HTTP", "socks": "SOCKS"}
+
+
+class DeleteLicense:
+    def __init__(self, idrac, module):
+        self.idrac = idrac
+        self.module = module
+
+    def execute(self, module):
+        license_id = module.params.get('license_id')
+        check_license_id(self, module, license_id, "delete")
+        license_url = get_license_url(self)
+        delete_license_url = license_url + f"/{license_id}"
+        delete_license_response = self.idrac.invoke_request(delete_license_url, 'DELETE')
+        status = delete_license_response.status_code
+        if status == 204:
+            module.exit_json(msg=SUCCESS_DELETE_MSG, changed=True)
+        else:
+            module.exit_json(FAILURE_MSG.format(operation="delete", license_id=license_id), failed=True)
+        return delete_license_response
+
+
+class ExportLicense:
+    STATUS_SUCCESS = [200, 202]
+
```

Comment:
Can't we declare this variable globally as it is being used in two classes?

Reply:
Done